### PR TITLE
apis: add Mid tier api

### PR DESCRIPTION
**Pull request by saintube**
### Ⅰ. Describe what this PR does

Add apis for node recommendation and Mid tier.

Proposed in #1385.

### Ⅱ. Does this pull request fix one issue?

Part of #1361.

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

**Review by jasonliu747 — COMMENTED** [review]
will there be a proposal for recommendation feature?

**Comment by saintube** [comment]
> will there be a proposal for recommendation feature?

Yes. The proposal will be submitted in another PR.

File: apis/extension/slo_controller_config.go [hunk 1]
```diff
@@ -199,13 +199,27 @@ type ColocationStrategy struct {
...
-	ColocationStrategyExtender     `json:",inline"`             // for third-party extension
+
+	CPUReclaimThresholdPercent    *int64           `json:"cpuReclaimThresholdPercent,omitempty" validate:"omitempty,min=0,max=100"`
+	MemoryReclaimThresholdPercent *int64           `json:"memoryReclaimThresholdPercent,omitempty" validate:"omitempty,min=0,max=100"`
+	MemoryCalculatePolicy         *CalculatePolicy `json:"memoryCalculatePolicy,omitempty"`
+	DegradeTimeMinutes            *int64           `json:"degradeTimeMinutes,omitempty" validate:"omitempty,min=1"`
+	UpdateTimeThresholdSeconds    *int64           `json:"updateTimeThresholdSeconds,omitempty" validate:"omitempty,min=1"`
+	ResourceDiffThreshold         *float64         `json:"resourceDiffThreshold,omitempty" validate:"omitempty,gt=0,max=1"`
+
+	MidCPUReclaimPercent        *int64 `json:"midCPUReclaimPercent,omitempty"`
```
Comment (zwzhang0107) [inline apis/extension/slo_controller_config.go:210]: better to Integrate into struct for related option: MidPriority{}, RecommenderPolicy{}

**Pull request merged**